server/methods/saveUserPreferences.js

```diff
@@ -50,11 +50,7 @@ Meteor.methods({
 			} else {
 				delete preferences.mergeChannels;
 			}
-			if (settings.unreadRoomsMode) {
```

Comment:
well it must be due to changes made again.
I will check all the changes again and let you know when it will be working perfectly.